abci/server/socket_server.go

```diff
@@ -153,7 +153,12 @@ func (s *SocketServer) handleRequests(closeConn chan error, conn io.Reader, resp
 		// make sure to recover from any app-related panics to allow proper socket cleanup
 		r := recover()
 		if r != nil {
-			closeConn <- fmt.Errorf("recovered from panic: %v", r)
+			const size = 64 << 10
+			buf := make([]byte, size)
+			buf = buf[:runtime.Stack(buf, false)]
+			err := fmt.Errorf("recovered from panic: %v\n%s", r, buf)
+			stdlog.Println(err)
```

Comment:
I think we only need to do this if `logger` is `nil`. I think output should also go to `stderr` instead of `stdout`, e.g. `fmt.Fprintf(os.Stderr, ...` or `log.New(os.StdErr).Printf(...)`